src/components/sidebar/helpers/generate-install-view-nav-items.ts

```diff
@@ -0,0 +1,39 @@
+import { MenuItem } from '../types'
+import { ProductData } from 'types/products'
+
+/**
+ * Given a `ProductData` object, generates the nav data for the `Sidebar` that
+ * appears on the associated Product's `ProductDownloads` page
+ * (/vault/downloads, /waypoint/downloads, etc.).
+ *
+ * Depends on the data located in the associated `src/data/${product.slug}.json`
+ * to be loaded into the `product` object passed.
+ *
+ * Can optionally be passed menuItems, which determine sidebar content.
+ * If menuItems are not provided, falls back to a default generated set
+ * identical to the items shown on product landing views.
+ */
+export const generateInstallViewNavItems = (
+  product: ProductData,
+  menuItems?: MenuItem[]
+) => {
+  const backToLinkProps = {
+    text: `${product.name} Home`,
+    href: `/${product.slug}`,
+  }
+  const levelButtonProps = {
+    levelUpButtonText: `${product.name} Home`,
+    levelDownButtonText: 'Previous',
+  }
+  const menuItemsWithFallback = menuItems ? menuItems : []
```

Comment:
🎨 I think this can be simplified to be `menuItems || []`, but totally up to you!

Reply:
Forgot to comment - thanks for this, updated! 💖 ✅